server/schedulers/hot_region.go

```diff
@@ -711,20 +705,40 @@ func (h *hotScheduler) selectDstStoreByScore(candidates map[uint64]struct{}, reg
 	} else {
 		storesScore = h.writeScores
 	}
-	if storesScore.Variation() > minScoreVariationLimit {
-		maxScore := storesScore.Max()                      // with sort
-		for _, scoreInfo := range storesScore.scoreInfos { // select from small to large
-			if scoreInfo.GetScore()+regionBytesRate < maxScore*hotRegionScheduleFactor {
-				dstStoreID := scoreInfo.GetStoreID()
-				if _, ok := candidates[dstStoreID]; ok {
-					return dstStoreID
-				}
+	maxScore := storesScore.Max()                      // with sort
+	for _, scoreInfo := range storesScore.scoreInfos { // select from small to large
+		if scoreInfo.GetScore()+regionBytesRate < maxScore*hotRegionScheduleFactor {
+			dstStoreID := scoreInfo.GetStoreID()
+			if _, ok := candidates[dstStoreID]; ok {
+				return dstStoreID
 			}
 		}
 	}
 	return 0
 }
 
+func (h *hotScheduler) checkVariation(src, dst uint64, regionBytesRate float64, balanceType BalanceType) bool {
```

Comment:
We don't need this function. The src/dst selecting algorithm can make sure this function always return true. What's more, in the future we may intentionally create operators breaks this check to balance more complex loads.

Reply:
how will we do to prevent overschedule?

Reply:
Actually, after fixing the unstable store statistics, there is no overschedule has been observed. So, I think it's not the problem we need to think now.